torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -584,6 +584,14 @@ def mark_sharding(t: Union[torch.Tensor, XLAShardedTensor], mesh: Mesh,
   assert len(t.shape) == len(partition_spec), \
     f"Partition spec length ({len(partition_spec)}) should be equal to the input rank ({len(t.shape)})."
 
+  tx = maybe_get_torchax()
+  if tx is not None and (t, tx.tensor.Tensor):
+    from jax.sharding import PartitionSpec, Mesh as JMesh, NamedSharding
+    device_ids = mesh.device_ids.reshape(mesh.mesh_shape)
+    jax_mesh = JMesh(device_ids, mesh.axis_names)
```

Comment:
Hmm, do you think the PyTorch/XLA device IDs will translate to JAX device IDs directly? I think the device IDs in the PyTorch/XLA mesh is PyTorch/XLA specific, like it's just an integer into some PyTorch/XLA global variable.

It should be safer to do what https://github.com/pytorch/xla/blob/cce7cc0782bbf507c96e1d81d705db8ec5099f0a/torch_xla/experimental/splash_attention.py#L67-L90 does